hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -653,37 +608,57 @@ private static void processRestoreMetadata(HoodieActiveTimeline metadataTableTim
    */
   public static Map<MetadataPartitionType, HoodieData<HoodieRecord>> convertMetadataToRecords(
       HoodieEngineContext engineContext, HoodieActiveTimeline metadataTableTimeline,
-      HoodieRollbackMetadata rollbackMetadata, MetadataRecordsGenerationParams recordsGenerationParams,
-      String instantTime, Option<String> lastSyncTs, boolean wasSynced) {
+      HoodieTableMetaClient dataTableMetaClient,
+      HoodieRollbackMetadata rollbackMetadata, String instantTime) {
     final Map<MetadataPartitionType, HoodieData<HoodieRecord>> partitionToRecordsMap = new HashMap<>();
-    Map<String, List<String>> partitionToDeletedFiles = new HashMap<>();
-    Map<String, Map<String, Long>> partitionToAppendedFiles = new HashMap<>();
 
-    List<HoodieRecord> filesPartitionRecords =
-        convertMetadataToRollbackRecords(metadataTableTimeline, rollbackMetadata, partitionToDeletedFiles, partitionToAppendedFiles, instantTime, lastSyncTs, wasSynced);
+    List<HoodieRecord> filesPartitionRecords = convertMetadataToRollbackRecords(rollbackMetadata, instantTime);
+
+    List<HoodieRecord> reAddedRecords = getHoodieRecordsForLogFilesFromRollbackPlan(dataTableMetaClient, instantTime);
+    filesPartitionRecords.addAll(reAddedRecords);
     final HoodieData<HoodieRecord> rollbackRecordsRDD = engineContext.parallelize(filesPartitionRecords, 1);
+
     partitionToRecordsMap.put(MetadataPartitionType.FILES, rollbackRecordsRDD);
 
     return partitionToRecordsMap;
   }
 
+  private static List<HoodieRecord> getHoodieRecordsForLogFilesFromRollbackPlan(HoodieTableMetaClient dataTableMetaClient, String instantTime) {
+    /*List<HoodieInstant> instants = dataTableMetaClient.reloadActiveTimeline().filterRequestedRollbackTimeline()
+        .filter(instant -> instant.getTimestamp().equals(instantTime) && instant.isRequested()).getInstants();*/
+
+    HoodieInstant rollbackInstant = new HoodieInstant(HoodieInstant.State.REQUESTED, HoodieTimeline.ROLLBACK_ACTION, instantTime);
+
+    // HoodieInstant rollbackInstant = instants.get(0);
+    HoodieInstant requested = HoodieTimeline.getRollbackRequestedInstant(rollbackInstant);
+    try {
+      HoodieRollbackPlan rollbackPlan = TimelineMetadataUtils.deserializeAvroMetadata(
+          dataTableMetaClient.getActiveTimeline().readRollbackInfoAsBytes(requested).get(), HoodieRollbackPlan.class);
+
+      Map<String, Map<String, Long>> partitionToLogFilesMap = new HashMap<>();
+
+      rollbackPlan.getRollbackRequests().forEach(rollbackRequest -> {
+        partitionToLogFilesMap.computeIfAbsent(rollbackRequest.getPartitionPath(), s -> new HashMap<>());
+        // fetch only log files that are expected to be RB'd in DT as part of this rollback. these log files will not be deleted, but rendered
+        // invalid once rollback is complete.
+        partitionToLogFilesMap.get(rollbackRequest.getPartitionPath()).putAll(rollbackRequest.getLogBlocksToBeDeleted());
+      });
```

Comment:
Can you confirm  the key of `rollbackRequest.getLogBlocksToBeDeleted()` is file group id or is it the log filename?

Reply:
it refers to full log file name. 
Sample rollback.requested 
```
avrocat /tmp/hudi_trips_mor/.hoodie/20230705160035431.rollback.requested
{"instantToRollback": {"HoodieInstantInfo": {"commitTime": "20230705160008665", "action": "deltacommit"}}, "RollbackRequests": {"array": [{"partitionPath": "americas/brazil/sao_paulo", "fileId": {"string": "0c717479-0609-4dd5-8ea4-f2edcf17986c-0"}, "latestBaseInstant": {"string": "20230705155904980"}, "filesToBeDeleted": [], "logBlocksToBeDeleted": {"map": {"file:/tmp/hudi_trips_mor/americas/brazil/sao_paulo/.0c717479-0609-4dd5-8ea4-f2edcf17986c-0_20230705155904980.log.2_0-81-113": -1}}}, {"partitionPath": "asia/india/chennai", "fileId": {"string": "27df1310-75f0-4353-9834-b0736fd3f6f3-0"}, "latestBaseInstant": {"string": "20230705155904980"}, "filesToBeDeleted": [], "logBlocksToBeDeleted": {"map": {"file:/tmp/hudi_trips_mor/asia/india/chennai/.27df1310-75f0-4353-9834-b0736fd3f6f3-0_20230705155904980.log.2_2-81-115": -1}}}, {"partitionPath": "americas/united_states/san_francisco", "fileId": {"string": "ff8f1c4b-c2e8-4f7d-b7ad-7ecb385a5f1d-0"}, "latestBaseInstant": {"string": "20230705155904980"}, "filesToBeDeleted": [], "logBlocksToBeDeleted": {"map": {"file:/tmp/hudi_trips_mor/americas/united_states/san_francisco/.ff8f1c4b-c2e8-4f7d-b7ad-7ecb385a5f1d-0_20230705155904980.log.2_1-81-114": -1}}}]}, "version": {"int": 1}}
```